development/compiling/compiling_for_android.rst

```diff
@@ -199,6 +199,30 @@ file in the ``bin\`` directory of your Godot source folder, so that the
 next time you build you will automatically have the custom templates
 referenced.
 
+.. _doc_compiling_for_android_creating_source_zip:
+
+Creating an Android source ZIP
+------------------------------
+
+To create an Android source ZIP (which is used for custom builds), run the
+following command at the root of the Godot source repository folder::
+
+    cd platform/android/java
+    # On Windows
+    .\gradlew zipCustomBuild
+    # On Linux and macOS
+    ./gradlew zipCustomBuild
+
+You can run this task before or after compiling export templates, as the
+generated ZIP does not contain compiled binaries. Once the ``zipCustomBuild``
+task has finished running, ``android_source.zip`` will be generated in the
+``bin/`` folder.
+
+To use this newly generated Android source ZIP in your projects, place it in the
+``templates/<version>`` folder of your Godot user data folder. See
+:ref:`doc_data_paths` for the location of this folder depending on your
+operating system.
```

Comment:
@reduz simplified this process on `master` in https://github.com/godotengine/godot/pull/50481.

@akien-mga That PR should probably be cherry-picked in the `3.x` branch.

Reply:
It is better in that it allows specifying the path, but it is still only accessible to the UI, which doesnt help CI/CD integration. This proposal will improve CI/CD integration of the android build system: https://github.com/godotengine/godot-proposals/issues/3128